### PR TITLE
support additionalTextEdits for autocomplete results

### PR DESCRIPTION
**Pull request by pfgithub**
In typescript, autocompleting something will create additionalTextEdits if it needs to do something like add an import

https://user-images.githubusercontent.com/6010774/221436283-9e13db4e-43e0-4f3f-8de0-01de435bd88a.mp4

Issues:

- This likely won't work if there are two additional edits because one will be inserted and then the next one will try to insert in the wrong place
- If you accept the autocomplete too fast, it won't work because the `completionItem/resolve` response hasn't come in yet

### CODE REVIEW

File: init.lua [hunk 1]
```diff
@@ -353,7 +353,12 @@ end
 ---@param doc core.doc
 ---@param text_edit table
 ---@return boolean True on success
-local function apply_edit(doc, text_edit)
+local function apply_edit(doc, text_edit, opts)
```
Comment (Guldoman) [inline init.lua:356]: I'd avoid the opts table and just add the parameter directly.

**Review by Guldoman — COMMENTED** [review]
Sorry it took so long to get to this PR.

> * This likely won't work if there are two additional edits because one will be inserted and then the next one will try to insert in the wrong place

I think this might be an issue even when the `textEdit` is to be inserted *after* the completion (however rare that might be), so we might have to implement something to translate the subsequent edits in any case.
It would also avoid the whole `select_result` thing.

**Comment by PerilousBooklet** [comment]
Should we close this, since #128 was merged?

**Comment by Guldoman** [comment]
Yep.
@pfgithub thank you for the contribution, it inspired #128.

**Pull request closed without merging**